### ISSUE TITLE
Make venv follow links to a specific python version

### ISSUE THREAD

**Issue by EdwinBennink** [pypa/virtualenv#2808]
**What's the problem this feature will solve?**

When a venv is created using e.g. `python3 -m venv myvenv`, it will create the link `myvenv/bin/python3 -> /usr/bin/python3` instead of `myvenv/bin/python3.10 -> /usr/bin/python3.10`.
When a new python version is installed, `/usr/bin/python3` will (likely) link to the new version and so will the venv. This is not the desired behavior.

**Describe the solution you'd like**

Make `venv` follow links when creating the venv. The links in the `bin` dir should point to a specific version, e.g. `myvenv/bin/python3.10 -> /usr/bin/python3.10`, no matter if python was called using `/usr/bin/python` , `/usr/bin/python3` or `/usr/bin/python3.10` .

This enables installing other python versions without breaking existing virtual environments. 

An alternative is to make users aware of the way this works and that they should be specific when creating the venv.

**Comment by esafak** [comment]
Does [--copies](https://virtualenv.pypa.io/en/latest/cli_interface.html#copies) solve your problem?

**Comment by esafak** [comment]
What do you think, @gaborbernat ?

**Comment by gaborbernat** [comment]
In theory I'm ok with this, but would like for someone to check if venv does this, and if not raise it there to so we can get alignment.

**Comment by esafak** [comment]
@EdwinBennink can you check at https://github.com/python/cpython (cf. https://github.com/python/cpython/blob/main/Lib/venv) or something?

**Comment by gaborbernat** [comment]
You would want to raise an issue there too 👍 and link to this one, please!

**Comment by pfmoore** [comment]
I can confirm that this is the behaviour in the core venv module:

```
$ ls -l /usr/bin/python3
lrwxrwxrwx 1 root root 10 Aug  7  2024 /usr/bin/python3 -> python3.12
$ /usr/bin/python3 -m venv example_venv --without-pip
$ ls -l example_venv/bin
total 24
-rwxrwxrwx 1 gustav gustav 9033 Jul 31 17:08 Activate.ps1
-rwxrwxrwx 1 gustav gustav 2068 Jul 31 17:08 activate
-rwxrwxrwx 1 gustav gustav  942 Jul 31 17:08 activate.csh
-rwxrwxrwx 1 gustav gustav 2221 Jul 31 17:08 activate.fish
lrwxrwxrwx 1 gustav gustav    7 Jul 31 17:07 python -> python3
lrwxrwxrwx 1 gustav gustav   16 Jul 31 17:07 python3 -> /usr/bin/python3
lrwxrwxrwx 1 gustav gustav    7 Jul 31 17:07 python3.12 -> python3
```

So this would need a feature request to core Python to get this changed. And if virtualenv wants to follow the core behaviour, which seems sensible to me, that feature request would need to be accepted and implemented in the stdlib first.

**Comment by esafak** [comment]
Feel free to make the request there @pfmoore and we can follow up here.

**Comment by pfmoore** [comment]
Nope, I'm not personally interested in this behaviour - I simply had a Unix system to hand so I checked as @gaborbernat requested. In actual fact, I'd argue *against* this feature in the stdlib, on backward compatibility grounds (it could be made optional, via a flag, I guess).

If someone else wants to raise this with core Python, go ahead.

**Comment by esafak** [comment]
What would you suggest doing then? @EdwinBennink have your thoughts changed?

**Comment by gaborbernat** [comment]
> So this would need a feature request to core Python to get this changed. And if virtualenv wants to follow the core behaviour, which seems sensible to me, that feature request would need to be accepted and implemented in the stdlib first.

I'm happy to implement it here first if we get venv stdlib buy-in for this change.

**Comment by pfmoore** [comment]
Me, personally? Nothing - I don't see the need for this change. I'll leave it to @EdwinBennink to argue the case that it's worth doing and that the compatibility break is worthwhile.

IMO,

> An alternative is to make users aware of the way this works and that they should be specific when creating the venv.

is sufficient.

**Comment by esafak** [comment]
How/when should we make users aware, then? Would a message after creating the virtualenv be acceptable?

**Comment by pfmoore** [comment]
In the documentation?
